File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
class Solution {
    int N;
    public void dfs(int x, int[][] computers, int[] visited){
        visited[x] = 1;
        for(int i = 0; i < N; i++){
            if(computers[x][i] == 1 && visited[i] == 0){
                dfs(i, computers, visited);
            }
        }
    }
    public int solution(int n, int[][] computers) {
        int answer = 0;
        N = n;
        int[] visited = new int[n];
        for(int i = 0; i < n; i++){
            if(visited[i] == 0){
                dfs(i, computers, visited);
                answer+= 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
N = 0
def solution(n, computers):
    global N
    N = n
    visited = [False] * n
    answer = 0
    for i in range(n):
        if not visited[i]:
            dfs(visited,computers,i)
            answer +=1           
    return answer

def dfs(visited, computers, x):
    visited[x] = True
    for i in range(0,N):
        if computers[x][i] == 1 and not visited[i]:
            dfs(visited, computers,i)

            
    